### PR TITLE
Resolve conflicts in otherlibs/

### PR DESCRIPTION
**Pull request by poechsel**
Wasn't able to get the ml files to compile as they are assuming the existence of `Mutex.t` in the stdlib

Most conflicts were in the Makefiles

### CODE REVIEW

File: otherlibs/dynlink/Makefile [hunk 1]
```diff
@@ -341,18 +266,8 @@ ifeq "$(strip $(NATDYNLINK))" "true"
 endif
 
 partialclean:
-<<<<<<< HEAD
 	rm -f $(extract_crc) *.cm[ioaxt] *.cmti *.cmxa \
```
Comment (ncik-roberts) [inline otherlibs/dynlink/Makefile:269]: `mergebase` has `$(extract_crc)` and `tip-5` removes it; should we remove it too?

**Review by ncik-roberts — APPROVED** [review]
This looks fine to me. I guess the changes to `dynlink_platform_intf.ml`, while fine, will conflict with the changes I made in https://github.com/ocaml-flambda/ocaml-jst/pull/210 to get other things to build. Someone will have to fix this in some PR, but I don't think it matters too much how they'll resolve it (as we'll fix this the "right way" when we can build with the new stdlib).

**Comment by ncik-roberts** [comment]
Also I see this doesn't resolve conflicts in these files:
  * `otherlibs/systhreads/st_posix.h`
  * `otherlibs/unix/dune`
  * `otherlibs/systhreads/st_stubs.c`

but I'm fine leaving these for another PR if you just want to merge this one.

File: otherlibs/dynlink/Makefile (hunk 1 above)
Comment (poechsel) [reply otherlibs/dynlink/Makefile]: Done

**Comment by poechsel** [comment]
`otherlibs/systhreads/st_stubs.c` seems a bit tricky to do so I wanted to do it in a further PR

**Comment by ncik-roberts** [comment]
Looks good, thanks.

**Pull request merged**